draft-ietf-quic-transport.md

```diff
@@ -3383,1364 +3804,966 @@ The STOP_SENDING frame is as follows:
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                        Crypto Data (*)                      ...
+|                                                               |
++                                                               +
+|                                                               |
++                   Stateless Reset Token (128)                 +
+|                                                               |
++                                                               +
+|                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ~~~
-{: #crypto-format title="CRYPTO Frame Format"}
 
-The CRYPTO frame contains the following fields:
+The fields are:
 
-Offset:
+Length:
 
-: A variable-length integer specifying the byte offset in the stream for the
-  data in this CRYPTO frame.
+: An 8-bit unsigned integer containing the length of the connection ID.  Values
+  less than 4 and greater than 18 are invalid and MUST be treated as a
+  connection error of type PROTOCOL_VIOLATION.
 
-Length:
+Sequence Number:
 
-: A variable-length integer specifying the length of the Crypto Data field in
-  this CRYPTO frame.
+: The sequence number assigned to the connection ID by the sender.  See
+  {{issuing-connection-ids}}.
 
-Crypto Data:
+Connection ID:
 
-: The cryptographic message data.
+: A connection ID of the specified length.
 
-There is a separate flow of cryptographic handshake data in each encryption
-level, each of which starts at an offset of 0. This implies that each encryption
-level is treated as a separate CRYPTO stream of data.
+Stateless Reset Token:
 
-Unlike STREAM frames, which include a Stream ID indicating to which stream the
-data belongs, the CRYPTO frame carries data for a single stream per encryption
-level. The stream does not have an explicit end, so CRYPTO frames do not have a
-FIN bit.
+: A 128-bit value that will be used for a stateless reset when the associated
+  connection ID is used (see {{stateless-reset}}).
 
+An endpoint MUST NOT send this frame if it currently requires that its peer send
+packets with a zero-length Destination Connection ID.  Changing the length of a
+connection ID to or from zero-length makes it difficult to identify when the
+value of the connection ID changed.  An endpoint that is sending packets with a
+zero-length Destination Connection ID MUST treat receipt of a NEW_CONNECTION_ID
+frame as a connection error of type PROTOCOL_VIOLATION.
 
-# Packetization and Reliability {#packetization}
+Transmission errors, timeouts and retransmissions might cause the same
+NEW_CONNECTION_ID frame to be received multiple times.  Receipt of the same
+frame multiple times MUST NOT be treated as a connection error.  A receiver can
+use the sequence number supplied in the NEW_CONNECTION_ID frame to identify new
+connection IDs from old ones.
```

Comment:
I think that we can remove the first two sentences of this paragraph.  They are vestiges of the design prior to us putting sequence numbers back in.